src/main/java/com/amihaiemil/eoyaml/ReadYamlSequence.java

```diff
@@ -145,10 +147,22 @@
                     || trimmed.endsWith(">")
                 ) {
                     kids.add(
-                        this.significant.toYamlNode(
-                            line, this.guessIndentation
-                        )
+                            this.significant.toYamlNode(
+                                    line, this.guessIndentation
+                            )
                     );
+                } else if (trimmed.matches("^-[ ]*\\{}")) {
+                    kids.add(new EmptyYamlMapping(new ReadYamlMapping(
+                            new RtYamlLine("", line.number()-1),
+                            this.all,
+                            this.guessIndentation
+                    )));
+                } else if (trimmed.matches("^-[ ]*\\[]")) {
+                    kids.add(new EmptyYamlSequence(new ReadYamlSequence(
+                            new RtYamlLine("", line.number()-1),
+                            this.all,
+                            this.guessIndentation
+                    )));
```

Comment:
@newmana Here you handle the reading of empty sequences or mappings within a read sequence. 

Looks ok, but I don't see any changes to the class ReadYamlMapping.java, which handles a read mapping. In other words, reading of empty sequences/mappings within a mapping will not work. 

Or am I missing something? :D

Reply:
You got it. I just found the same thing.